cps/kobo.py

```diff
@@ -392,6 +394,222 @@ def get_metadata(book):
     return metadata
 
 
+@kobo.route("/v1/library/tags", methods=["POST"])
+@login_required
+# Creates a Shelf with the given items, and returns the shelf's uuid.
+def HandleTagCreate():
+    shelf_request = request.json
+    name, items = None, None
+    try:
+        name = shelf_request["Name"]
+        items = shelf_request["Items"]
+    except KeyError:
+        log.debug("Received malformed v1/library/tags request.")
+        abort(400, description="Malformed tags POST request. Data is missing 'Name' or 'Items' field")
+
+    shelf = ub.session.query(ub.Shelf).filter(and_(ub.Shelf.name) == name, ub.Shelf.user_id ==
+                                              current_user.id).one_or_none()
```

Comment:
I think the parthesis around `and_(ub.Shelf.name)` make no sense, should be around whole filter request?

Reply:
I'm removing the and_ clause from all these filter calls in 06c15a7 .
The reason this works is because the filter function joins together it's argument into an and_ clause by default.